yarn-project/simulator/src/client/execution_result.ts

```diff
@@ -20,6 +28,14 @@ export interface NoteAndSlot {
   noteTypeId: Fr;
 }
 
+export class CountedLog<TLog extends UnencryptedL2Log | EncryptedL2Log> implements IsEmpty {
+  constructor(public log: TLog, public counter: number) {}
+
+  isEmpty(): boolean {
+    return !this.log.data.length && !this.counter;
+  }
+}
+
```

Comment:
This implements `IsEmpty` so I can take advantage of the existing method `sortByCounter`. Happy to move to a new file if it's cleaner.